src/resolve/resolved-link.c

```diff
@@ -726,7 +726,8 @@ DnsServer* link_set_dns_server(Link *l, DnsServer *s) {
         dns_server_unref(l->current_dns_server);
         l->current_dns_server = dns_server_ref(s);
 
-        if (l->unicast_scope)
+        /* Skip flushing the cache if server stale feature is enabled. */
+        if (l->unicast_scope && l->manager->stale_retention_usec == 0)
```

Comment:
this should not be part of the test commit, of course

Reply:
hmm i think this simply fix will actually be fine. but could you document that in the man page. i.e. where the StaleRetentionSec= option is described, simply say "If this option is enabled, cache flushes when changing servers are disabled."

Reply:
updated man with "If this option is enabled, the cache will not be flushed when changing servers."